src/js/program/ad-program-controller.js

```diff
@@ -113,6 +113,7 @@ export default class AdProgramController extends ProgramController {
             event.oldstate = model.get(PLAYER_STATE);
             adMediaModelContext.set('mediaState', event.newstate);
         });
+        this.off();
         this.on('fullscreenchange', this._nativeFullscreenHandler, this);
```

Comment:
You really should listening to events on `this`. It means you are ignoring the the code path that actually triggers the event.

Instead either add a check for 'fullscreenchange' in the `provider.on('all'` handler, or add a new event listener on the provider or source(s) of the event itself.

In events.js
`NATIVE_FULLSCREEN = 'fullscreenchange'`

Reply:
Note that this event should be routed through the router.So if it's not, that is what you should be fixing.